### PR TITLE
Fixes for NXP T1024 with Integrity OS

### PR DESCRIPTION
**Pull request by dgarske**
Fixes and improvements to get Integrity OS booting on NXP T1024:
* Added additional FDT fixups for FMAN, SDHC, Ethernet, PCI. 
* Added NXP T1024 mEMAC, MDIO and PHY setup.
* Pulled in latest PCI code from TigerLake efforts
* Added PCI init and enumeration
* Added MRAM init
* Fixed issue with CPLD due to incorrect AMASK (also in T2080 port)
* Fix for CPU core clock calculation.

### CODE REVIEW

**Comment by dgarske** [comment]
@danielinux this is a good state to merge. Fixes some bugs and adds features. I expect another to eventually to resolve a few outstanding issues.

File: hal/nxp_t1024.c [hunk 1]
```diff
@@ -1609,9 +1990,601 @@ static int fman_upload_firmware(const struct qe_firmware *firmware)
...
+    #ifdef DEBUG_PHY
+        val = hal_phy_read_indirect(phydev, DP83867_RGMIIDCTL, MDIO_DEVAD_NONE);
+        wolfBoot_printf("RGMIIDCTL delay=0x%x\n", val);
+    #endif
+
+#if DP83867_IO_MUX_CFG_IO_IMPEDANCE_MIN >= 0
+    #ifdef DEBUG_PHY
+        wolfBoot_printf("Impedance Match 0x%x\n", DP83867_IO_MUX_CFG_IO_IMPEDANCE_MIN);
+    #endif
+    #ifdef DEBUG_PHY
```
Comment (danielinux) [inline hal/nxp_t1024.c:2510]: Group under same ifdef?

File: src/pci.c [hunk 2]
```diff
@@ -498,16 +529,25 @@ static int pci_program_bar(uint8_t bus, uint8_t dev, uint8_t fun,
...
 
     vid = pci_config_read16(bus, dev, fun, PCI_VENDOR_ID_OFFSET);
     did = pci_config_read16(bus, dev, fun, PCI_DEVICE_ID_OFFSET);
 
-    PCI_DEBUG_PRINTF("PCI enum: dev: %x:%x.%x vid: %x did: %x\r\n",
+    PCI_DEBUG_PRINTF("\r\n\r\nPCI: %x:%x.%x %x:%x\r\n",
                     bus, dev, fun, (int)vid, (int)did);
 }
+#else
+static inline void pci_dump_id(uint8_t bus, uint8_t dev, uint8_t fun)
```
Comment (danielinux) [inline src/pci.c:544]: this could be 
`#define pci_dump_id(...) do{}while(0)`  to save one symbol allocation

File: hal/nxp_t1024.c [hunk 3]
```diff
@@ -1609,9 +1990,601 @@ static int fman_upload_firmware(const struct qe_firmware *firmware)
...
+        default:
+            break;
+    }
+    return "Unknown";
+}
+
+static const char* hal_phy_vendor_str(uint32_t id)
+{
+    switch (id) {
+        case 0x2000a231:
```
Comment (danielinux) [inline hal/nxp_t1024.c:2337]: value is worth a define

File: hal/nxp_t1024.c [hunk 4]
```diff
@@ -1609,9 +1990,601 @@ static int fman_upload_firmware(const struct qe_firmware *firmware)
...
+
+    /* Set the on-board RGMII PHY addresses */
+    phydevs[FM1_DTSEC4].interface = PHY_INTERFACE_MODE_RGMII;
+    phydevs[FM1_DTSEC4].phyaddr = RGMII_PHY1_ADDR;
+    phydevs[FM1_DTSEC3].interface = PHY_INTERFACE_MODE_RGMII;
+    phydevs[FM1_DTSEC3].phyaddr = RGMII_PHY2_ADDR;
+
+    reg = get32(DCFG_RCWSR(4));
+    reg = (reg & RCWSR4_SRDS1_PRTCL) >> RCWSR4_SRDS1_PRTCL_SHIFT;
+    if (reg == 0x95) {
```
Comment (danielinux) [inline hal/nxp_t1024.c:2556]: Value is worth a define

**Review by danielinux — CHANGES_REQUESTED** [review]
Looks good. Some minor changes for readability

**Review by danielinux — APPROVED** [review]

**Pull request merged**